code/game/objects/structures/tram_walls.dm

```diff
@@ -96,3 +96,232 @@
+	slicing_duration = 200   //diamond wall takes twice as much time to slice
+	max_integrity = 800
+	explosion_block = 3
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = list(SMOOTH_GROUP_CLOSED_TURFS, SMOOTH_GROUP_WALLS, SMOOTH_GROUP_DIAMOND_WALLS)
+	canSmoothWith = list(SMOOTH_GROUP_DIAMOND_WALLS)
+	custom_materials = list(/datum/material/diamond = 4000)
+
+/obj/structure/tramwall/bananium
+	name = "bananium wall"
+	desc = "A wall with bananium plating. Honk!"
+	icon = 'icons/turf/walls/bananium_wall.dmi'
+	icon_state = "bananium_wall-0"
+	base_icon_state = "bananium_wall"
+	mineral = /obj/item/stack/sheet/mineral/bananium
+	tram_wall_type = /obj/structure/tramwall/bananium
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = list(SMOOTH_GROUP_CLOSED_TURFS, SMOOTH_GROUP_WALLS, SMOOTH_GROUP_BANANIUM_WALLS)
+	canSmoothWith = list(SMOOTH_GROUP_BANANIUM_WALLS)
+	custom_materials = list(/datum/material/bananium = 4000)
+
+/obj/structure/tramwall/sandstone
+	name = "sandstone wall"
+	desc = "A wall with sandstone plating. Rough."
+	icon = 'icons/turf/walls/sandstone_wall.dmi'
+	icon_state = "sandstone_wall-0"
+	base_icon_state = "sandstone_wall"
+	mineral = /obj/item/stack/sheet/mineral/sandstone
+	tram_wall_type = /obj/structure/tramwall/sandstone
+	explosion_block = 0
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = list(SMOOTH_GROUP_CLOSED_TURFS, SMOOTH_GROUP_WALLS, SMOOTH_GROUP_SANDSTONE_WALLS)
+	canSmoothWith = list(SMOOTH_GROUP_SANDSTONE_WALLS)
+	custom_materials = list(/datum/material/sandstone = 4000)
+
+/obj/structure/tramwall/uranium
+	article = "a"
+	name = "uranium wall"
+	desc = "A wall with uranium plating. This is probably a bad idea."
+	icon = 'icons/turf/walls/uranium_wall.dmi'
+	icon_state = "uranium_wall-0"
+	base_icon_state = "uranium_wall"
+	mineral = /obj/item/stack/sheet/mineral/uranium
+	tram_wall_type = /obj/structure/tramwall/uranium
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = list(SMOOTH_GROUP_CLOSED_TURFS, SMOOTH_GROUP_WALLS, SMOOTH_GROUP_URANIUM_WALLS)
+	canSmoothWith = list(SMOOTH_GROUP_URANIUM_WALLS)
+	custom_materials = list(/datum/material/uranium = 4000)
+
+	var/active = null
+	var/last_event = 0
+
+/obj/structure/tramwall/uranium/attackby(obj/item/W, mob/user, params)
+	radiate()
+	return ..()
+
+/obj/structure/tramwall/uranium/attack_hand(mob/user, list/modifiers)
+	radiate()
+	. = ..()
+
+/obj/structure/tramwall/uranium/proc/radiate()
+	if(!active)
+		if(world.time > last_event+15)
+			active = 1
+			radiation_pulse(
+				src,
+				max_range = 2,
+				threshold = RAD_LIGHT_INSULATION,
+				chance = URANIUM_IRRADIATION_CHANCE,
+				minimum_exposure_time = URANIUM_RADIATION_MINIMUM_EXPOSURE_TIME,
+			)
+			for(var/turf/closed/wall/mineral/uranium/wall in orange(1,src))
+				wall.radiate()
+			for(var/obj/structure/falsewall/uranium/falsewall in orange(1,src))
+				falsewall.radiate()
+			for(var/obj/structure/tramwall/uranium/tramwall in orange(1,src))
+				tramwall.radiate()
```

Comment:
Seeing as you're doing this twice, do my last advice for making these signals, but then make a shared proc for this (since signals should only be fired from one place).

Reply:
any suggestions in which file the shared proc should go?